hoot-core/src/main/cpp/hoot/core/io/OsmApiChangesetElement.h

```diff
@@ -340,26 +343,26 @@ class XmlRelation : public XmlElement
 
 private:
   /** List of relation members */
-  QList<XmlMember> _members;
+  QList<ChangesetRelationMember> _members;
 };
 /** Handy typedef for relation shared pointer */
-typedef std::shared_ptr<XmlRelation> XmlRelationPtr;
+typedef std::shared_ptr<ChangesetRelation> ChangesetRelationPtr;
 
 /** Custom sorting function to sort IDs from -1 to -n followed by 1 to m */
+bool id_sort_order(long lhs, long rhs);
 class osm_id_sort
 {
 public:
   bool operator() (long lhs, long rhs) const
   {
-    if (lhs > 0 && rhs > 0)       return lhs < rhs; //  Positive numbers count up
-    else if (lhs < 0 && rhs < 0)  return lhs > rhs; //  Negative numbers count down
-    else                          return lhs < rhs; //  Negative numbers come before positive
+    return id_sort_order(lhs, rhs);
   }
 };
```

Comment:
I extracted the ID sorting to a function that is called within the sorting object so that both `std::map` and `std::vector` objects could be sorted with the same logic.